port/esp32/adapter/src/ipadapter.c

```diff
@@ -877,7 +878,7 @@ network_event_thread(void *data)
       OC_DBG("interface index %d", message->endpoint.interface_index);
       OC_DBG("%s", "");
 
-#if defined(OC_DYNAMIC_ALLOCATION) && !defined(OC_INOUT_BUFFER_SIZE)
+#ifdef OC_HAS_FEATURE_MESSAGE_DYNAMIC_BUFFER
       void *tmp = realloc(
         message->data,
         message->length +
```

Comment:
We can get rid of the +1, since we removed the write from `coap_oscore_parse_options`

Reply:
Removed because shrink is done by the func `oc_network_receive_event(message);`